spyder_kernels/customize/spyderpdb.py

```diff
@@ -203,75 +206,12 @@ def default(self, line):
-                    # The idea here is that the best way to emulate being in a
-                    # function is to actually execute the code in a function.
-                    # A function called `_spyderpdb_code` is created and
-                    # called. It will first load the locals, execute the code,
-                    # and then update the locals.
-                    #
-                    # One limitation of this approach is that locals() is only
-                    # a copy of the curframe locals. This means that closures
-                    # for example are early binding instead of late binding.
-
-                    # Create a function
-                    indent = "    "
-                    code = ["def _spyderpdb_code():"]
-
-                    # Add locals in globals
-                    # If the debugger is recursive, the globals could already
-                    # have a _spyderpdb_locals as it might be shared between
-                    # levels
-                    if "_spyderpdb_locals" in globals:
-                        globals["_spyderpdb_locals"].append(locals)
-                    else:
-                        globals["_spyderpdb_locals"] = [locals]
-
-                    # Load locals if they have a valid name
-                    # In comprehensions, locals could contain ".0" for example
-                    code += [indent + "{k} = _spyderpdb_locals[-1]['{k}']".format(
-                        k=k) for k in locals if k.isidentifier()]
-
-                    # The code comes here
-
-                    # Update the locals
-                    code += [indent + "_spyderpdb_locals[-1].update("
-                             "__spyder_builtins__.locals())"]
-
-                    # Run the function
-                    code += ["_spyderpdb_code()"]
-
-                    # Parse the function
-                    fun_ast = ast.parse('\n'.join(code) + '\n')
-
-                    # Inject code_ast in the function before the locals update
-                    fun_ast.body[0].body = (
-                        fun_ast.body[0].body[:-1]  # The locals
-                        + code_ast.body  # Code to run
-                        + fun_ast.body[0].body[-1:]  # Locals update
-                    )
-                    code_ast = fun_ast
-
-                try:
-                    exec(compile(code_ast, "<stdin>", "exec"), globals)
-                finally:
-                    if locals is not globals:
-                        # CLeanup code
-                        globals.pop("_spyderpdb_code", None)
-                        if len(globals["_spyderpdb_locals"]) > 1:
-                            del globals["_spyderpdb_locals"][-1]
-                        else:
-                            del globals["_spyderpdb_locals"]
-                        
+                if is_magic:
+                    # For runcell and all magic using locals
+                    # Because the locals() can not be modified
```

Comment:
```suggestion
                    # For runcell and all magics using locals. That's because
                    # the locals() dict can not be modified.
```